File: sorting/java/SelectSort.java
class SelectSort
{
    // Java implementation of select sort.
    public static void main(String[] args)
    {
        // randomly sorted list
        int[] nums = {87, 76, 81, 53, 11};
        int smallest;
        int temp;

        for (int i=0; i<nums.length - 1; i++)
        {
            // find the smallest element in the array
            smallest = i;
            for (int j=i+1; j < nums.length; j++)
            {
                if (nums[j] < nums[smallest])
                    smallest = j;
            }

            if (smallest != i)
            {
                // swap nums[i] and nums[smallest]
                temp = nums[smallest];
                nums[smallest] = nums[i];
                nums[i] = temp;
            }
        }
        // print out the result
        for (int i=0; i<nums.length; i++)
            System.out.print(nums[i] + " ");
        System.out.println("");
    }
}

File: sorting/ruby/select_sort.rb
# select sort
# An in place sorting method that selects the correct item from the unsorted
# part of the list and places it at the end of the sorted section of the list.

def select_sort(x)

    smallest = 0

    x.length().times do |index|
        smallest = index
        (index..x.length()-1).each do |check|
            if x[check] < x[smallest]
                smallest = check
            end
        end
        temp = x[index]
        x[index] = x[smallest]
        x[smallest] = temp
    end

    return x
end
